builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,231 @@
+		ReplicationGroupDescription:	aws.String(description),
+		CacheNodeType:            	aws.String(cacheNodeType),
+		AutomaticFailoverEnabled: 	aws.Boolean(automaticFailover),
+		NumCacheClusters:         	aws.Long(int64(numCacheClusters)),
+		Engine:				aws.String(engine),
+		EngineVersion:			aws.String(engineVersion),
+	}
+
+	if hasParameterGroup {
+		req.CacheParameterGroupName = aws.String(parameterGroupName.(string))
+	}
+
+	_, err := conn.CreateReplicationGroup(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache replication group: %s", err)
+	}
+
+	pending := []string{"creating"}
+	stateConf := &resource.StateChangeConf{
+		Pending:    pending,
+		Target:     "available",
+		Refresh:    ReplicationGroupStateRefreshFunc(conn, d.Id(), "available", pending),
+		Timeout:    15 * time.Minute,
+		Delay:      20 * time.Second,
+		MinTimeout: 5 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for state to become available: %v", d.Id())
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for elasticache (%s) to be created: %s", d.Id(), sterr)
+	}
+
+	d.SetId(replicationGroupId)
+
+	return nil
+}
+
+func resourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+	req := &elasticache.DescribeReplicationGroupsInput{
+		ReplicationGroupID: aws.String(d.Id()),
+	}
+
+	res, err := conn.DescribeReplicationGroups(req)
+	if err != nil {
+		return err
+	}
+
+	if len(res.ReplicationGroups) == 1 {
+		c := res.ReplicationGroups[0]
+		d.Set("replication_group_id", c.ReplicationGroupID)
+		d.Set("description", c.Description)
+		d.Set("automatic_failover", c.AutomaticFailover)
+	}
+
+	return nil
+}
+
+func resourceAwsElasticacheReplicationGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	req := &elasticache.ModifyReplicationGroupInput{
+		ApplyImmediately:   aws.Boolean(true),
+		ReplicationGroupID: aws.String(d.Id()),
+	}
+
+	if d.HasChange("description") {
+		description := d.Get("description").(string)
+		req.ReplicationGroupDescription = aws.String(description)
+	}
+
+	if d.HasChange("automatic_failover") {
+		automaticFailover := d.Get("automatic_failover").(bool)
+		req.AutomaticFailoverEnabled = aws.Boolean(automaticFailover)
+	}
+
+	_, err := conn.ModifyReplicationGroup(req)
+	if err != nil {
+		d.Partial(true)
+		return fmt.Errorf("Error updating Elasticache replication group: %s", err)
+	}
+
+	return resourceAwsElasticacheReplicationGroupRead(d, meta)
+}
+
+func resourceAwsElasticacheReplicationGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	req := &elasticache.DeleteReplicationGroupInput{
+		ReplicationGroupID: aws.String(d.Id()),
+	}
+
+	_, err := conn.DeleteReplicationGroup(req)
+	if err != nil {
+		return fmt.Errorf("Error deleting Elasticache replication group: %s", err)
```

Comment:
same comment here about checking the error, if it's "not found" let's go ahead and set the ID to `""` and not return an error.